bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -0,0 +1,218 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+/**
+ * The {@link PushbulletHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+@NonNullByDefault
+public class PushbulletHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PushbulletHandler.class);
+
+    private final Gson gson = new GsonBuilder().create();
+
+    private static final Version VERSION = FrameworkUtil.getBundle(PushbulletHandler.class).getVersion();
+
+    private static final Pattern CHANNEL_PATTERN = Pattern.compile("^[a-zA-Z0-9_-]+$");
+
+    private @Nullable PushbulletConfiguration config;
+
+    public PushbulletHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("About to handle {} on {}", command, channelUID);
+
+        // Future improvement: If recipient is already set, send a push on a command channel change
+        // check reconnect channel of the unifi binding for that
+
+        logger.debug("The Pushbullet binding is a read-only binding and cannot handle command '{}'.", command);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(PushbulletConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Name and Token are both "required", so set the Thing immediately ONLINE.
+        updateStatus(ThingStatus.ONLINE);
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(PushbulletActions.class);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String message, String type) {
+        return sendPush(recipient, "", message, type);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String title, @Nullable String message, String type) {
+        boolean result = false;
+
+        logger.debug("sendPush is called for ");
+        logger.debug("Thing {}", thing);
+        logger.debug("Thing Label: '{}'", thing.getLabel());
```

Comment:
Wouldn't this be resolved by the toString of the argument of the above logline

Reply:
In principle, yes. I just like being showed line-by-line debug information. Consider it a matter of taste please :)